tensorflow/core/kernels/non_max_suppression_op.cu.cc

```diff
@@ -529,16 +534,31 @@ Status DoNMS(OpKernelContext* context, const Tensor& boxes,
   }
   Tensor* output_indices = nullptr;
   int num_outputs = std::min(num_to_keep, output_size);  // no padding!
-  TF_RETURN_IF_ERROR(
-      context->allocate_output(0, TensorShape({num_outputs}), &output_indices));
-  if (num_outputs == 0) return Status::OK();
+  if (pad_to_max_output && num_outputs!=max_output_size) {
```

Comment:
Please remove line 427 (alias max_output_size). It's confusing to have both output_size and max_output_size meaning the same thing.